src/client/clientmap.h

```diff
@@ -87,10 +86,18 @@ class ClientMap : public Map, public scene::ISceneNode
 
 	void updateCamera(const v3f &pos, const v3f &dir, f32 fov, const v3s16 &offset)
 	{
+		v3s16 previous_block = getContainerPos(floatToInt(m_camera_position + intToFloat(m_camera_offset, BS), BS), MAP_BLOCKSIZE);
+
 		m_camera_position = pos;
 		m_camera_direction = dir;
 		m_camera_fov = fov;
 		m_camera_offset = offset;
+
+		v3s16 current_block = getContainerPos(floatToInt(m_camera_position + intToFloat(m_camera_offset, BS), BS), MAP_BLOCKSIZE);
+
+		// reorder the blocks when camera crosses block boundary
+		if (previous_block != current_block)
+			updateDrawList();
```

Comment:
This is usually only called from `Game::updateFrame`.
Now we could potentially call `updateDrawlist` twice from main game loop if both camera direction and position have changed, right? If so, might be better to fold this update into the logic in `Game::updateFrame`.